internal/eventstore/v3/push.go

```diff
@@ -14,47 +14,57 @@ import (
 	"github.com/zitadel/logging"
 
 	"github.com/zitadel/zitadel/internal/eventstore"
+	"github.com/zitadel/zitadel/internal/telemetry/tracing"
 	"github.com/zitadel/zitadel/internal/zerrors"
 )
 
 func (es *Eventstore) Push(ctx context.Context, commands ...eventstore.Command) (events []eventstore.Event, err error) {
+	ctx, span := tracing.NewSpan(ctx)
+	defer func() { span.EndWithError(err) }()
+
 	tx, err := es.client.BeginTx(ctx, nil)
 	if err != nil {
 		return nil, err
 	}
+
+	defer func() {
+		if err != nil {
+			rollbackErr := tx.Rollback()
+			logging.OnError(rollbackErr).Info("rollback failed")
+			return
+		}
+		err = tx.Commit()
+	}()
+
 	// tx is not closed because [crdb.ExecuteInTx] takes care of that
 	var (
 		sequences []*latestSequence
 	)
 
-	err = crdb.ExecuteInTx(ctx, &transaction{tx}, func() error {
```

Comment:
Wouldn't we loose the automatic retries for serilizable transactions?

Reply:
I reset that one